### PR TITLE
RSDK-7045 Implement StreamTicks 

### PR DESCRIPTION
**Pull request by oliviamiller**
Implements the StreamTicks board API. also tested with code samples client script.

### CODE REVIEW

File: lib/src/components/board/board.dart [hunk 1]
```diff
@@ -58,6 +70,12 @@ abstract class Board extends Resource {
   /// Return the current value of the interrupt which is based on the type of Interrupt.
   Future<int> digitalInterruptValue(String digitalInterruptName, {Map<String, dynamic>? extra});
 
+  // Stream digital interrupts ticks.
+  Future<ResponseStream<StreamTicksResponse>> streamTicks(List<String> interrupts, {Map<String, dynamic>? extra});
```
Comment (njooma) [inline lib/src/components/board/board.dart:74]: Can we make the return type a `Stream` of `Tick`s please. Take a look at how we do it for [`tailLogs`](https://github.com/viamrobotics/viam-flutter-sdk/blob/337e9d2767eec68c022bca94a6f3c852b94b822b/lib/src/app/app.dart#L95) to get an idea of how to convert a `ResponseStream` to a native `Stream`

File: lib/src/components/board/service.dart [hunk 2]
```diff
@@ -112,8 +114,23 @@ class BoardService extends BoardServiceBase {
...
-    // TODO: implement streamTicks
-    throw UnimplementedError();
+  Stream<StreamTicksResponse> streamTicks(ServiceCall call, StreamTicksRequest request) async* {
+    final board = _fromManager(request.name);
+
+    final ticks = Queue<Tick>();
+    await board.addCallbacks(request.pinNames, ticks);
+
+    try {
+      while (true) {
```
Comment (njooma) [inline lib/src/components/board/service.dart:124]: Does this ever end? I don't see a break ever... Not sure if it's supposed to end ever but just wondering

**Review by njooma — CHANGES_REQUESTED** [review]

File: lib/src/components/board/service.dart (hunk 2 above)
Comment (oliviamiller) [reply lib/src/components/board/service.dart:124]: No its supposed to continue forever until the client cancels it.

File: lib/src/components/board/client.dart [hunk 3]
```diff
@@ -120,6 +123,22 @@ class BoardClient extends Board implements ResourceRPCClient {
...
+  Future<void> addCallbacks(List<String> interrupts, Queue<Tick> tickQueue, {Map<String, dynamic>? extra}) async {
+    // addCallbacks not implemented on client side since it is a helper for StreamTicks.
+    throw UnimplementedError();
+  }
+
+  @override
+  Stream<Tick> streamTicks(List<String> interrupts, {Map<String, dynamic>? extra}) {
+    final response = client.streamTicks(StreamTicksRequest()
+      ..name = name
+      ..pinNames.addAll(interrupts));
```
Comment (stuqdog) [inline lib/src/components/board/client.dart:136]: looks like we're not passing along `extra` here, is that intentional?

**Review by stuqdog — COMMENTED** [review]
Generally looks good to me! modulo question about `extra`

File: lib/src/components/board/client.dart (hunk 3 above)
Comment (oliviamiller) [reply lib/src/components/board/client.dart]: not intentional thanks for catching! added

**Review by njooma — APPROVED** [review]

**Review by stuqdog — APPROVED** [review]

**Pull request merged**